src/app/vc-receipt-status/middleware.ts

```diff
@@ -6,40 +6,49 @@ import { isAxiosError } from "axios";
 import config from "../../config/config";
 import { logger } from "../../lib/logger";
 
-enum AppVcReceiptStatus {
+export enum AppVcReceiptStatus {
   COMPLETED = "COMPLETED",
   ERROR = "ERROR",
   PROCESSING = "PROCESSING",
   INTERVENTION = "INTERVENTION",
 }
 
-export const getAppVcReceiptStatus: RequestHandler = async (
+export const getAppVcReceipt = async (
   req: Request,
-  res: Response,
-) => {
+): Promise<AppVcReceiptStatus> => {
   try {
-    // For browser tests
-    if (config.ENABLE_PREVIEW && process.env.NODE_ENV === "local") {
-      res.status(200).json({ status: AppVcReceiptStatus.COMPLETED });
-      return;
-    }
-
     const appVcResponse = await appVcReceived(req);
     if (!isJourneyResponse(appVcResponse.data)) {
       throw new Error(
         "Journey response expected from successful check app vc receipt response.",
       );
     }
-
     req.session.journey = appVcResponse.data.journey;
-    res.status(200).json({ status: AppVcReceiptStatus.COMPLETED });
-    return;
+
+    return AppVcReceiptStatus.COMPLETED;
   } catch (error) {
     if (isAxiosError(error) && error.response?.status === 404) {
-      res.status(200).json({ status: AppVcReceiptStatus.PROCESSING });
-      return;
+      return AppVcReceiptStatus.PROCESSING;
     }
     logger.error(error, "Error getting app vc receipt status");
-    res.status(500).json({ status: AppVcReceiptStatus.ERROR });
+    return AppVcReceiptStatus.ERROR;
+  }
+};
+
+export const getAppVcReceiptStatus: RequestHandler = async (
+  req: Request,
+  res: Response,
+) => {
+  if (config.ENABLE_PREVIEW && process.env.NODE_ENV === "local") {
+    res.status(200).json({ status: AppVcReceiptStatus.COMPLETED });
+    return;
   }
+
+  const status = await getAppVcReceipt(req);
```

Comment:
Can we tidy up the naming a little, maybe `getAppVcReceipt` -> `getAppVcReceiptStatus` given it's getting the status, and maybe the handler `pollVcReceiptStatus` ?

Reply:
`getAppVcReceiptStatus` could even be `getAppVcReceiptStatusAndStoreJourneyResponse` so as not to hide its side effect